src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs

```diff
@@ -1,270 +1,101 @@
-        {
-            jsObject.AssertNotDisposed();
-
-            if (shouldAddInflight != 0)
+            catch (Exception ex)
             {
-                jsObject.AddInFlight();
+                arg_exc.ToJS(ex);
             }
-            return jsObject.JSHandle;
         }
 
         [MethodImplAttribute(MethodImplOptions.NoInlining)] // https://github.com/dotnet/runtime/issues/71425
-        public static IntPtr TryGetCSOwnedObjectJSHandleRef(in object rawObj, int shouldAddInflight)
+        public static void CreateTaskCallback(JSMarshalerArgument* arguments_buffer)
         {
-            JSObject? jsObject = rawObj as JSObject;
-            if (jsObject != null && shouldAddInflight != 0)
-            {
-                jsObject.AddInFlight();
-            }
-            return jsObject?.JSHandle ?? IntPtr.Zero;
-        }
-
-        [MethodImplAttribute(MethodImplOptions.NoInlining)] // https://github.com/dotnet/runtime/issues/71425
-        public static void CreateCSOwnedProxyRef(IntPtr jsHandle, JSHostImplementation.MappedType mappedType, int shouldAddInflight, out JSObject jsObject)
-        {
-            JSObject? res = null;
-
-            lock (JSHostImplementation.s_csOwnedObjects)
+            ref JSMarshalerArgument arg_exc = ref arguments_buffer[0];
+            ref JSMarshalerArgument arg_return = ref arguments_buffer[1];
+            try
             {
-                if (!JSHostImplementation.s_csOwnedObjects.TryGetValue((int)jsHandle, out WeakReference<JSObject>? reference) ||
-                    !reference.TryGetTarget(out res) ||
-                    res.IsDisposed)
-                {
-#pragma warning disable CS0612 // Type or member is obsolete
-                    res = mappedType switch
-                    {
-                        JSHostImplementation.MappedType.JSObject => new JSObject(jsHandle),
-                        JSHostImplementation.MappedType.Array => new Array(jsHandle),
-                        JSHostImplementation.MappedType.ArrayBuffer => new ArrayBuffer(jsHandle),
-                        JSHostImplementation.MappedType.DataView => new DataView(jsHandle),
-                        JSHostImplementation.MappedType.Function => new Function(jsHandle),
-                        JSHostImplementation.MappedType.Uint8Array => new Uint8Array(jsHandle),
-                        _ => throw new ArgumentOutOfRangeException(nameof(mappedType))
-                    };
-#pragma warning restore CS0612 // Type or member is obsolete
-                    JSHostImplementation.s_csOwnedObjects[(int)jsHandle] = new WeakReference<JSObject>(res, trackResurrection: true);
-                }
+                JSHostImplementation.TaskCallback holder = new JSHostImplementation.TaskCallback();
+                arg_return.slot.Type = MarshalerType.Object;
+                arg_return.slot.GCHandle = JSHostImplementation.GetJSOwnedObjectGCHandle(holder);
```

Comment:
Do we need `arg_exc.Initialize();` here?

Reply:
No, the caller on the JS side already done that. We are receiving initialized `arguments_buffer` 
Good thinking thought, keep it coming!